twisted/twisted
# #11585 Update the PR template and dev docs to not require a specific merge commit message.

## Scope and purpose

Fixes #11585

With the GitHub integration, a specific commit message is no longer required.

GitHub can use the PR title as the commit message.
And if the PR description contains a reference to a ticket, it is auto closed.


## Changes

This updates the PR template not to request the commit message.
The PR template is updated to talk about the PR title.
The PR template was updated with the latest links.

I have removed the contribution checklist as I think they are just noise.
We should use the check list for real tasks associated with a PR.
The CI checks should automatically force you to "check" those steps.

I have updated the CI to fail if the PR title has no ticket reference.
It is a very simple check.

As a drive-by I have move the docs into a separate job as docs + mypy was taking to long.
Also narrative docs now builds apidocs. So there is no need for both.
More work for docs CI is at https://github.com/twisted/twisted/pull/11578

## Trade-off

The PR title check runs on commit push. In this way we have it in a single YAML file.
The problem is that if the PR is wrong, you will need a new commit to have the PR title check pass.


The other option is to have the PR title check in a separate YAML file that runs on every PR title update.


## How to test

See a failure here https://github.com/twisted/twisted/runs/7401209770?check_suite_focus=true#step:8:24

----------

Testing the PR template is complicated...as it will be visible on the next run
Below is the same content to review the MD rendering

-----------

## Scope and purpose

Fixes #

Add a few words about why this PR is needed and what is its scope.
You can just refer to the associated ticket(s).

Add any comments about trade-offs (if any) made in this PR and the reasoning behind them.

Add mentions of things that are not covered here and are planed to be done in separate PRs.


## Contributor Checklist:

This process apply to *all* pull requests, no matter how small.
Have a look at [our developer documentation](https://docs.twisted.org/en/latest/core/development/dev-process.html) before submitting your Pull Request.

Below is a non-exaustive list (as a reminder):

* The title of the PR should describe the changes and starts with the associated issue number (including the `#` character for GitHub auto-links).
* A release notes news fragment file was create in src/twisted/newsfragments/ (see: [Release notes fragments docs.](https://docs.twisted.org/en/latest/core/development/dev-process.html#release-notes-management))
* The automated tests were updated.
* Once all checks are green, request a review by leaving a comment that contains the `please review` words.
  This will trigger the review process.
